### ISSUE TITLE
ERC-1271 (potentially ERC-6492) Account Ownership Verification

### ISSUE THREAD

**Issue by kevtechi** [monerium/smart-contracts#39]
Hi,

Kevin from [Citizen Wallet](https://citizenwallet.xyz) here. 

We use account abstraction (ERC4337) wallets for everything.  

We were trying to set up Wallet Connect. Signing the "personal_sign" payload and sending it back to Monerium by using the address of the private key works. But attempting to sign by providing the address of our Smart Contract Wallet fails. We tried following a [ERC191](https://eips.ethereum.org/EIPS/eip-191) version `0x00` signature, but the check always seems to use ecrecover.

![Screenshot 2023-10-05 at 12 15 26](https://github.com/monerium/smart-contracts/assets/122159642/276ebdf0-1862-44be-852c-0124795c2e19)

Is there something that I'm missing or do you not support Smart Contract Wallets?

**Comment by kerma** [comment]
hi @kev-techi ! We do support https://eips.ethereum.org/EIPS/eip-1271 however the amount of different contracts we've been able to test with has been limited so far. The main use case was to support multi-sigs. 

Please try with posting "0x" as the signature (as per https://monerium.dev/api-docs#operation/profile-addresses multi signature address payload). Given you smart contract address implements 1271 it could work.

**Comment by kev-techi** [comment]
Ok, I'll give it a shot!

**Comment by kerma** [comment]
We actually do support smart contract 65 byte signatures in our contracts. But our backend cannot handle them correctly atm. We'll look into enabling this soon as well.

**Comment by kev-techi** [comment]
Ok, will you update here when it's enabled?

**Comment by kerma** [comment]
@kev-techi Can you give me an example address to check with? I just need the address part on any of our supported networks.

**Comment by gislik** [comment]
It’s actually already working for either a single signer or a 1-of-N multisig wallets

Here’s a [link](https://app.safe.global/transactions/messages?safe=gor:0x419A88bcD86248803668daf3eB52A8CCB2bb58C4)  to an example using a Safe wallet on Görli. 

![IMG_2157](https://github.com/monerium/smart-contracts/assets/943039/837187a0-129e-4b19-999a-14573e5e8504)

**Comment by gislik** [comment]
Here’s a [link](https://app.safe.global/transactions/messages?safe=gor:0x419A88bcD86248803668daf3eB52A8CCB2bb58C4)  to the API docunentation to add a smart wallet. Make sure that you include the “chain” and “network” parameters. 
![Uploading IMG_2158.jpeg…]()

**Comment by kev-techi** [comment]
Hey, I'm doing this through Wallet Connect. I'm not calling your API directly, just responding to the session and signing requests I receive. This triggers your frontend to make those API calls automatically. 

<img width="1805" alt="Screenshot 2023-10-06 at 23 12 35" src="https://github.com/monerium/smart-contracts/assets/122159642/d6db69ca-c089-4643-9b66-557f117fccc7">

I get a "personal_sign" request which I need to sign and send back. I can only send back a single string. This should be the signed version of what you sent me.

Normally, there should be an auth request from your side which I can respond to with a structured signature.

Hence my feeling that this flow was missing from your Wallet Connect implementation.

**Comment by kev-techi** [comment]
When I scan this from our wallet and approve the session, Monerium shows a "Connect wallet" page:

![Screenshot 2023-10-06 at 23 25 38](https://github.com/monerium/smart-contracts/assets/122159642/08554403-5d7f-44fe-a140-7f147c952291)

The wallet address at the top is the smart contract wallet. Now, when I press the "Connect" button, I receive this:

```
{id: 1696627566032919, jsonrpc: 2.0, method: wc_sessionRequest, params: {request: {method: personal_sign, params: [0x4920686572656279206465636c6172652074686174204920616d207468652061646472657373206f776e65722e, 0x0eB87B2b50183FDa651130c9543E8b822F3ff2A7]}, chainId: eip155:1}}
```

"personal_sign" expect that I sign `params[0]` and send it back. When I send it back, the "Connect wallet" page makes the API call you referenced in your documentation. It calls `https://monerium.app/api/emoney/profiles/{profileId}/addresses` POST with this payload:

```
{
    "message": "I hereby declare that I am the address owner.",
    "address": "0x0eB87B2b50183FDa651130c9543E8b822F3ff2A7",
    "network": "mainnet",
    "chain": "ethereum",
    "signature": "0x12383418b499e8cc5fec663eb81dc8973dbe655c95a44e1d1a5b141a9237bfa27af5d8739b93cd17b6b7d266dc24038e2310351204a178673666df63ce2c1dfb2c",
    "accounts": [
        {
            "currency": "eur",
            "chain": "polygon",
            "network": "mainnet",
            "isVisible": true
        },
        {
            "currency": "eur",
            "chain": "ethereum",
            "network": "mainnet",
            "isVisible": true
        }
    ]
}
```

And your backend responds with:

```
{
    "code": 400,
    "status": "Bad Request",
    "message": "Invalid signature",
    "details": {
        "address": "0x0eB87B2b50183FDa651130c9543E8b822F3ff2A7",
        "message": "address recovered from signature does not match given address",
        "recovered": "0xafdbee37289279a160665da4acc573d37d965a9e"
    }
}
```

![Screenshot 2023-10-06 at 23 26 31](https://github.com/monerium/smart-contracts/assets/122159642/abe4af40-ee0d-475f-bb76-2d0e52174bb6)

**Comment by kev-techi** [comment]
If I set the address as the private key's address (and not the smart contract wallet), it works. From this, I know that I am following the Wallet connect docs correctly.

I can see from here that I should receive an Auth Request rather than "personal_sign". This would allow me to respond with something like this:

```
await web3Wallet.respondAuthRequest(
  id: args.id,
  iss: 'did:pkh:eip155:1:0x06C6A22feB5f8CcEDA0db0D593e6F26A3611d5fa',
  signature: CacaoSignature(t: CacaoSignature.EIP191, s: sig),
);
```

**Comment by kev-techi** [comment]
Wallet Connect supports Smart Contract Wallets and so does your API, so it might be just tweaking that "Connect wallet" page to send the proper request when I hit "Connect".

**Comment by kerma** [comment]
@kev-techi maybe a stupid question, but why you say https://etherscan.io/address/0x0eB87B2b50183FDa651130c9543E8b822F3ff2A7 is a smart contract address? There is no code associated with it on chain, so to me it looks like any other externally owned address.

**Comment by kev-techi** [comment]
I was giving an example just to explain the flow 🙂.

It is here: https://polygonscan.com/address/0x0eB87B2b50183FDa651130c9543E8b822F3ff2A7#tokentxns

**Comment by kerma** [comment]
Ok, that explains. What I can tell so far is that:
* 0x0eB87B2b50183FDa651130c9543E8b822F3ff2A7 is not a smart contract on ethereum mainnet
* 0x0eB87B2b50183FDa651130c9543E8b822F3ff2A7 is smart contract on polygon
* The error you received is due to the fact that the payload that reaches our backend includes the address as ethereum mainnet, not polygon. 

Two hints here, frontend connected address looks to be ethereum:
<img width="472" alt="image" src="https://github.com/monerium/smart-contracts/assets/735108/e0f03d2c-1024-43ad-8694-7967d3e851c9">

Payload includes eth mainnet (because it is using the connected wallet for information):
<img width="932" alt="image" src="https://github.com/monerium/smart-contracts/assets/735108/4a4d5f13-07f0-4c8e-9b45-9ac1303c1986">

Can you switch the network once you land on the signature page or pick the polygon already while in your wallet? Atm it looks like you start your connection on eth mainnet, but you actually want to link polygon address.

**Comment by kev-techi** [comment]
In your wallet connect configuration, Ethereum is mandatory and Polygon is optional.

**Comment by kev-techi** [comment]
It would be great if I could simply only link the network I want.

**Comment by gislik** [comment]
Hey I noticed something similar when I was trying to link the Safe wallet.

Can you do me a favor? When you reach this screen, can you reload the page. Once I reloaded it changed page to only care about the network that I wanted. If you can confirm that to work for you then, we’d  be able to fix on our end.

![](https://user-images.githubusercontent.com/122159642/273335583-abe4af40-ee0d-475f-bb76-2d0e52174bb6.png)

By the way. Is there a way for me to replicate what you’re doing on my side?

**Comment by kev-techi** [comment]
A `wc_sessionPropose` jsonrpc request is sent as soon as I scan the Wallet Connect QR code. This is Monerium specifying what rpc methods and chains it supports.

It always specifies a single required name space `eip155:1` with methods `[eth_sendTransaction, personal_sign]`. Polygon and Gnosis are mentioned as optional.

These are the params:

```
{requiredNamespaces: {eip155: {chains: [eip155:1], methods: [eth_sendTransaction, personal_sign], events: [chainChanged, accountsChanged], rpcMap: {1: https://cloudflare-eth.com}}}, optionalNamespaces: {eip155: {chains: [eip155:137, eip155:100], methods: [eth_sendTransaction, personal_sign, eth_accounts, eth_requestAccounts, eth_sendRawTransaction, eth_sign, eth_signTransaction, eth_signTypedData, eth_signTypedData_v3, eth_signTypedData_v4, wallet_switchEthereumChain, wallet_addEthereumChain, wallet_getPermissions, wallet_requestPermissions, wallet_registerOnboarding, wallet_watchAsset, wallet_scanQRCode], events: [chainChanged, accountsChanged, message, disconnect, connect], rpcMap: {1: https://cloudflare-eth.com, 100: https://rpc.gnosischain.com, 137: https://polygon-rpc.com}}}, relays: [{protocol: irn}], proposer: {publicKey: 56f4e78a18f0ec25c04bd8455dbb72d06a24e5caefa06fee4d359b7a86f4ae44, metadata: {description: , url: https://monerium.app, icons: [https://monerium.app/icon.png], name: Accounts | Monerium}}}
```


It would be great if you could only request the relevant chain: https://specs.walletconnect.com/2.0/specs/clients/sign/namespaces

**Comment by kev-techi** [comment]
> Hey I noticed something similar when I was trying to link the Safe wallet.
> 
> Can you do me a favor? When you reach this screen, can you reload the page. Once I reloaded it changed page to only care about the network that I wanted. If you can confirm that to work for you then, we’d be able to fix on our end.
> 
> ![](https://user-images.githubusercontent.com/122159642/273335583-abe4af40-ee0d-475f-bb76-2d0e52174bb6.png)
> 
> By the way. Is there a way for me to replicate what you’re doing on my side?

reloading does not change anything since the request happens before this screen.

I tried it anyway and it still always defaults to ethereum.

**Comment by kev-techi** [comment]
Alright, I deployed an Account Factory and generated an ERC4337 Account Contract with 1271 verification: [0xC19Ee7B6E4204dcA883D4EB8449Ee5F25cf0a0E4](https://etherscan.io/address/0xc19ee7b6e4204dca883d4eb8449ee5f25cf0a0e4)  

When I use this, the choices are then limited to only Ethereum on Monerium. 

![Screenshot 2023-10-09 at 00 35 02](https://github.com/monerium/smart-contracts/assets/122159642/811efa3b-3958-4fda-85e1-36054a1a21ca)

I was then able to sign as that contract and the connection was accepted.

It still shows all 3 chains in the dashboard.

![Screenshot 2023-10-09 at 00 47 43](https://github.com/monerium/smart-contracts/assets/122159642/81044885-9bea-4edb-a803-bf2659eeeb2f)


This is great, because it shows that it's possible 🎉 . 

There is still the issue where it's not possible for me to do this for Polygon. I would have to deploy a contract under the same address on Ethereum as on Polygon and then sign on Ethereum. 

Is it possible to fix this so that I can select and sign for Polygon?

**Comment by kev-techi** [comment]
Source code for the Account: https://github.com/citizenwallet/smartcontracts/blob/main/contracts/accounts/Account.sol

Source code for the Account Factory: https://github.com/citizenwallet/smartcontracts/blob/main/contracts/accounts/AccountFactory.sol

**Comment by kev-techi** [comment]
Hey @gislik, will you guys be able to look into a fix for the frontend of Monerium here? 

It's going to be quite heavy to make Etherium Contract Wallets as well for people on Polygon.

**Comment by einaralex** [comment]
I'm looking into this @kev-techi .
We are using RainbowKit which abstracts all of the WC namespace management, I should be able to detect the chainId in the wallet before rendering the modal. I'll do some testig.

**Comment by einaralex** [comment]
@kev-techi Quick question. Before you sign, are you unable to choose the chain logo in the upper left corner and choose Polygon, before clicking the Link button so that the signature request will be performed on Polygon?

I know this is not a solution to your problem, but should allow you to continue. Let me know and then we'll map out the next steps.

**Comment by kev-techi** [comment]
<img width="844" alt="Screenshot 2023-10-14 at 15 04 45" src="https://github.com/monerium/smart-contracts/assets/122159642/ac4f86f7-5da7-4f38-98d9-69e13b6c324d">

Chain selection only comes after the QR code. The session proposal and approval are already sent back and forth with the wrong chain at that point. 

<img width="1444" alt="Screenshot 2023-10-14 at 15 20 22" src="https://github.com/monerium/smart-contracts/assets/122159642/db8684d4-c814-4988-854d-179ddaf7e77a">

But if I try switching chains after, it does the trick. Then it connects!

Thanks for the tip @einaralex .


We would still appreciate there being some way that is more intuitive to be able to decide "Polygon" for the user. Either we send something to you when we scan the QR code (through Wallet Connect) or the user interface on Monerium guides them to pick a chain first (if you go with the second option and there is a route "/accounts/new?chain=137", we can then open that page). 

We're onboarding non-crypto people here, so all these terms are foreign to them. Therefore, we're trying to remove as much complexity and crypto-speak as possible.

**Comment by kev-techi** [comment]
> We're onboarding non-crypto people here, so all these terms are foreign to them. Therefore, we're trying to remove as much complexity and crypto-speak as possible.

The chain switcher is also just an icon and no one from the non-crypto world knows what that icon means.

**Comment by kev-techi** [comment]
After the successful Wallet Connect connection, is it possible to get the IBAN number? Now that a Monerium account is configured, it would be great to show people the IBAN they can transfer to top receive EURe.

**Comment by kev-techi** [comment]
@einaralex any updates on the frontend fix?

**Comment by einaralex** [comment]
Hey @kev-techi,  I _plan_ to fix this in our downtime, we are currently in an active sprint cycle.
Regarding displaying the IBAN, we are also in the process of updating our UI to make it more focused on the IBAN.
Sorry for the late reply, there have been a lot of incidents recently with our growing customer base.

**Comment by kev-techi** [comment]
That makes sense, thanks for the response!

**Comment by JonasBoury** [comment]
Is this already fixed now?

**Comment by einaralex** [comment]
@kev-techi I removed Ethereum from being the enforced chain, it is still considered the required chain according to WC V2, but it should allow Polygon or Gnosis if the provider reads that from the wallet when connecting. Could you try again?


@JonasBoury I think the our new smart contracts are ready, but still in the process of being audited before we release them

**Comment by kev-techi** [comment]
Great news! I will check again.

**Comment by einaralex** [comment]
Did it work as expected?

**Comment by kev-techi** [comment]
Hey @einaralex , I just tried it the new interface (sorry for the delay) and it now shows "Polygon" correctly as the selected chain.

However, the flow that was working previously seems to not be working anymore. I can get up to this point when a `personal_sign` request is sent:
 
![Screenshot 2024-03-12 at 00 59 05](https://github.com/monerium/smart-contracts/assets/122159642/d9972b14-0193-45ca-a977-dfd97304e7af)

But after signing the message and returning it, I don't get any response (no error, no success).

I tested my hash + signature pair and they seem to be correctly constructed (I am able to ecrecover properly).

I also tested directly against the isValidSignature function on my Smart Contract Account and it returns the proper magic value `0x1626ba7e`.

You can try here:
hash: `0xb77c35c892a1b24b10a2ce49b424e578472333ee8d2456234fff90626332c50f`
signature: `0x146be9fec7bd23c43c5c1d7377484657d21629be2bf8431ce798ea6b243e0d485f04a48d271a8f46e60b18d5920b93fa5882f06aeffb3c5a3a4e88f1b08f81981b`

https://polygonscan.com/address/0x260965c708773E5754b5234d940AE8AbE0DF3d39#readProxyContract

The contract's owner is `0xFcC70e5bCF685cD1342063144d4dcb6922c2c438`, this can be recovered from the hash + signature.

I'm not sure what the issue could be. Is it possible to expose the error that is occurring?

**Comment by kev-techi** [comment]
If I return an invalid signature on purpose (I add an extra letter for example), I still don't receive any errors.

**Comment by einaralex** [comment]
@kev-techi 
Just to be sure, you are building on top of Safe, right?
We just noticed a few days ago that something recently changed that disabled the Safe on-chain signing.

**What is happening today:** when trying to sign, an off-chain signature is requested; if all owners sign, a signature hash is acquired, and the request to our API is sent off. This does not work because we do not support off-chain signatures yet, and when using smart contract wallets, our backend always expects the signature to be `0x`

**How it should work:** When a signature request is proposed, the request to our API is sent with the signature as `0x`, and it will respond with a 202 Status.
When all owners have signed, the account will be created.

I've been working on a[ fix for this](https://gist.github.com/einaralex/f07a41e46413461cbd04be6700be0165), and it's almost thoroughly tested and ready to be released to production ideally this week. I'll notify you when it's ready.

**Comment by kev-techi** [comment]
This is an on-chain 4337 Smart Contract Account with ERC1271 support (https://eips.ethereum.org/EIPS/eip-1271) . It's not a Safe. 

It was working a few months ago from your end, I was able to link it. And as you can see, the contract that I provided responds correctly to the hash + signature combination.

So, this could be another issue than the one you mentioned above.

We use ERC4337 account abstraction, this type of signature verification is necessary for our users to be able to use Monerium.

The main issue here is that there is no error message from Wallet Connect or from the Monerium interface. So, it's impossible to debug.

**Comment by kev-techi** [comment]
And this is indeed a signature with `0x`.

**Comment by kev-techi** [comment]
Sorry, don't mean to come off as ranting. Just clarifying some points related to your response. 

Hopefully we can resolve this.

**Comment by einaralex** [comment]
Hey @kev-techi , thanks for the explanation!
I was able to connect this wallet by using the signature you shared.

I'm making changes so not all smart contract addresses are treated the same, yours should be sending the actual signature as signature to the EMS. Hopefully I can release it by the end of today, I'll keep you posted.

**Comment by einaralex** [comment]
Changes are live now, can you please test again 🙏

**Comment by kev-techi** [comment]
<img width="1895" alt="Screenshot 2024-03-21 at 10 34 23" src="https://github.com/monerium/smart-contracts/assets/122159642/3606056f-46f2-4aea-825c-0213a176f9a8">

Same issue, I am requested to sign. 

I sign the way I mentioned above with a valid signature. And then nothing else happens.

_I ecRecover locally before sending to double-check and it seems ok._

**Comment by kev-techi** [comment]
@einaralex  I would highly recommend you trying to connect with a Smart Contract Account that is 1271 enabled in order to debug this better. Not a Gnosis Safe, but a 4337 account (or try using the Safe 4337 module). 

Not that 4337 should make a difference to a 1271 implementation, but at least you'll be able to debug this faster rather than me testing this for you.

This is the new account abstraction standard that all new EVM accounts are using. The whole ecosystem is moving to account contracts as the primary way of interacting with chains.

Coinbase launched their new wallet with 4337 as well. 

It has been months now of trying to integrate with Monerium and we would really like to start using the service, launching it in our app and promoting it. This really needs to be a higher priority.

**Comment by kev-techi** [comment]
We can also set up a call to get it working. I'm sure that synchronously, we could get this resolved much faster.

**Comment by einaralex** [comment]
Yeah this would be over quickly if I could get something to test wit.

**Comment by kev-techi** [comment]
Do you not have a 1271 account contract setup to test internally?

**Comment by kev-techi** [comment]
It worked on 14/10/2024, just a few messages up: https://github.com/monerium/smart-contracts/issues/39#issuecomment-1762921436 

We're still using the same ERC1271 compliant smart contract through Wallet Connect.

**Comment by einaralex** [comment]
Yeah, ok, I understand; the problem with that setup is that it broke other wallets like Argent. 
I'm trying to find a more permanent solution for what we have now, but I'll do some patching, so it should be working by tomorrow afternoon.

**Comment by kev-techi** [comment]
Ok, here is a full example:

Account: 0x260965c708773E5754b5234d940AE8AbE0DF3d39

Needs to be signed: 0x4920686572656279206465636c6172652074686174204920616d207468652061646472657373206f776e65722e

Hash: 0x18f9abb68d26b9d0e47af9ceff924d13559a0905c3d931ee4c86c123b57ac70e

Signature: 0x5a6e2be870587ef2648ed58201a349673ac70a50527c330b5f81f97c6696ab3a34ffe58908a609635b0450a152311598eac10a6223cfb843efb269649d41955b1c

Test here: https://polygonscan.com/address/0x260965c708773E5754b5234d940AE8AbE0DF3d39#readProxyContract

Trigger `isValidSignature` with the hash and signature, it will provide the magic value as expected.

Returns the magic value: `0x1626ba7e`

Follows this: https://eips.ethereum.org/EIPS/eip-1271
And this: https://docs.walletconnect.com/api/sign/smart-contract-wallet-usage

**Comment by kev-techi** [comment]
> Yeah, ok, I understand; the problem with that setup is that it broke other wallets like Argent. I'm trying to find a more permanent solution for what we have now, but I'll do some patching, so it should be working by tomorrow afternoon.

Ah, just as I was replying 😀.

Thanks for looking into this.

**Comment by einaralex** [comment]
Getting the network calls would still be excellent; I had this info.

The setup in our front end for signatures is not much different today than on the 24th of October.
So, I'm starting to run out of options.

I suspect the issue might be something changed with our wallet dependencies.

It would be extremely helpful if you could use this minimal setup to connect the wallet sign and tell me the result.
If nothing happens after signing, then we can know for sure that that's the problem.

If a signature is displayed on the screen, please share it.

https://3sth7y.csb.app/
Code available here: [Codesandbox](https://codesandbox.io/p/sandbox/rainbowkit-sign-3sth7y?file=%2Fpackage.json%3A12%2C22&layout=%257B%2522sidebarPanel%2522%253A%2522EXPLORER%2522%252C%2522rootPanelGroup%2522%253A%257B%2522direction%2522%253A%2522horizontal%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522id%2522%253A%2522ROOT_LAYOUT%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522clu95920d00063b6u9rnc41nk%2522%252C%2522sizes%2522%253A%255B100%252C0%255D%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522EDITOR%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522id%2522%253A%2522clu95920c00023b6uly5uw3ch%2522%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522SHELLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522id%2522%253A%2522clu95920c00033b6uay6hvuko%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522DEVTOOLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522id%2522%253A%2522clu95920c00053b6uelufjjuh%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%252C%2522sizes%2522%253A%255B54.83974333169628%252C45.16025666830372%255D%257D%252C%2522tabbedPanels%2522%253A%257B%2522clu95920c00023b6uly5uw3ch%2522%253A%257B%2522id%2522%253A%2522clu95920c00023b6uly5uw3ch%2522%252C%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clu96g0le00023b6um53w7235%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522FILE%2522%252C%2522filepath%2522%253A%2522%252Fsrc%252FWalletProvider.js%2522%252C%2522state%2522%253A%2522IDLE%2522%257D%252C%257B%2522id%2522%253A%2522clu97ich600023b6u8o9614t9%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522FILE%2522%252C%2522initialSelections%2522%253A%255B%257B%2522startLineNumber%2522%253A12%252C%2522startColumn%2522%253A22%252C%2522endLineNumber%2522%253A12%252C%2522endColumn%2522%253A22%257D%255D%252C%2522filepath%2522%253A%2522%252Fpackage.json%2522%252C%2522state%2522%253A%2522IDLE%2522%257D%255D%252C%2522activeTabId%2522%253A%2522clu97ich600023b6u8o9614t9%2522%257D%252C%2522clu95920c00053b6uelufjjuh%2522%253A%257B%2522id%2522%253A%2522clu95920c00053b6uelufjjuh%2522%252C%2522activeTabId%2522%253A%2522clu95m1br006e3b6uo7dmc0ww%2522%252C%2522tabs%2522%253A%255B%257B%2522type%2522%253A%2522UNASSIGNED_PORT%2522%252C%2522port%2522%253A0%252C%2522id%2522%253A%2522clu95m1br006e3b6uo7dmc0ww%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522path%2522%253A%2522%252F%2522%257D%255D%257D%252C%2522clu95920c00033b6uay6hvuko%2522%253A%257B%2522tabs%2522%253A%255B%255D%252C%2522id%2522%253A%2522clu95920c00033b6uay6hvuko%2522%257D%257D%252C%2522showDevtools%2522%253Atrue%252C%2522showShells%2522%253Afalse%252C%2522showSidebar%2522%253Atrue%252C%2522sidebarPanelSize%2522%253A21.61824051654561%257D)

I understand that this is starting to get frustrating and believe me, I want this to be over a.s.a.p. too.

The reality is we are working in a fast-moving ecosystem where doing changes or upgrading dependencies to fix one thing almost always introduces new bugs.

**Comment by einaralex** [comment]
Thank you!

**Comment by kev-techi** [comment]
If the Monerium Wallet Connect integration could expose the internal error that occurs, I could send you those or check to see what is wrong.

**Comment by kev-techi** [comment]
> Unfortunately, at least not yet, we can not catch them with the wallet provider we use. I've been in discussion with their team about it.
> 
> But it should output an error in the devTools console. Have you check there?

No errors, which is why I assume that they are being caught somewhere.

**Comment by kev-techi** [comment]
> Getting the network calls would still be excellent; I had this info.
> 
> What I thought the issue was, does not seem to be it either. The setup in our front end for signatures is not much different today than on the 24th of October. So, I'm starting to run out of options.
> 
> I suspect the issue might be something changed with our wallet dependencies.
> 
> It would be extremely helpful if you could use this minimal setup to connect the wallet sign and tell me the result. If nothing happens after signing, then we can know for sure that that's the problem.
> 
> If a signature is displayed on the screen, please share it.
> 
> https://3sth7y.csb.app/ Code available here: [Codesandbox](https://codesandbox.io/p/sandbox/rainbowkit-sign-3sth7y?file=%2Fpackage.json%3A12%2C22&layout=%257B%2522sidebarPanel%2522%253A%2522EXPLORER%2522%252C%2522rootPanelGroup%2522%253A%257B%2522direction%2522%253A%2522horizontal%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522id%2522%253A%2522ROOT_LAYOUT%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522clu95920d00063b6u9rnc41nk%2522%252C%2522sizes%2522%253A%255B100%252C0%255D%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522EDITOR%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522id%2522%253A%2522clu95920c00023b6uly5uw3ch%2522%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522SHELLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522id%2522%253A%2522clu95920c00033b6uay6hvuko%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522DEVTOOLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522id%2522%253A%2522clu95920c00053b6uelufjjuh%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%252C%2522sizes%2522%253A%255B54.83974333169628%252C45.16025666830372%255D%257D%252C%2522tabbedPanels%2522%253A%257B%2522clu95920c00023b6uly5uw3ch%2522%253A%257B%2522id%2522%253A%2522clu95920c00023b6uly5uw3ch%2522%252C%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clu96g0le00023b6um53w7235%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522FILE%2522%252C%2522filepath%2522%253A%2522%252Fsrc%252FWalletProvider.js%2522%252C%2522state%2522%253A%2522IDLE%2522%257D%252C%257B%2522id%2522%253A%2522clu97ich600023b6u8o9614t9%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522FILE%2522%252C%2522initialSelections%2522%253A%255B%257B%2522startLineNumber%2522%253A12%252C%2522startColumn%2522%253A22%252C%2522endLineNumber%2522%253A12%252C%2522endColumn%2522%253A22%257D%255D%252C%2522filepath%2522%253A%2522%252Fpackage.json%2522%252C%2522state%2522%253A%2522IDLE%2522%257D%255D%252C%2522activeTabId%2522%253A%2522clu97ich600023b6u8o9614t9%2522%257D%252C%2522clu95920c00053b6uelufjjuh%2522%253A%257B%2522id%2522%253A%2522clu95920c00053b6uelufjjuh%2522%252C%2522activeTabId%2522%253A%2522clu95m1br006e3b6uo7dmc0ww%2522%252C%2522tabs%2522%253A%255B%257B%2522type%2522%253A%2522UNASSIGNED_PORT%2522%252C%2522port%2522%253A0%252C%2522id%2522%253A%2522clu95m1br006e3b6uo7dmc0ww%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522path%2522%253A%2522%252F%2522%257D%255D%257D%252C%2522clu95920c00033b6uay6hvuko%2522%253A%257B%2522tabs%2522%253A%255B%255D%252C%2522id%2522%253A%2522clu95920c00033b6uay6hvuko%2522%257D%257D%252C%2522showDevtools%2522%253Atrue%252C%2522showShells%2522%253Afalse%252C%2522showSidebar%2522%253Atrue%252C%2522sidebarPanelSize%2522%253A21.61824051654561%257D)
> 
> I understand that this is starting to get frustrating and believe me, I want this to be over a.s.a.p. too.
> 
> The reality is we are working in a fast-moving ecosystem where doing changes or upgrading dependencies to fix one thing almost always introduces new bugs.

At least, you have a working minimal setup to continue now.

**Comment by kev-techi** [comment]
Hey @einaralex, any updates on the bugfix?

**Comment by kev-techi** [comment]
@gislik ?

**Comment by einaralex** [comment]
Hey, I've been cleaning up the code and tracing it back to how it was, and I've been trying to get it out; it's just currently stuck waiting on a bugfix in the backend,
It's still a shot in the dark, let's hope for the best

**Comment by kev-techi** [comment]
You don't sound super confident about the fix 😅

**Comment by kev-techi** [comment]
It still doesn't work, still no error exposed. 

Let's set up a call.

**Comment by kev-techi** [comment]
@einaralex , I'm also happy to submit a PR to make a fix to your dashboard.

**Comment by kev-techi** [comment]
That way, it will go faster to have both running at the same time and make the fix.

**Comment by kevtechi** [comment]
Hi @einaralex, any updates on the bug fix for ERC1271 verification?

**Comment by einaralex** [comment]
I successfully and effortlessly linked a [ERC-1271 wallet](https://polygonscan.com/address/0xB384c57268b65f7Cdfe1c547d6bE62dB105E7A4C#readProxyContract), through https://sequence.app/wallet, so that's not the issue.

Payload:
```
{
"message":"I hereby declare that I am the address owner.",
"address":"0xB384c57268b65f7Cdfe1c547d6bE62dB105E7A4C",
"chain":"polygon",
"signature":"0x00050000000000028e7f88ff485233293d8ee7ee6ee611c4764d6ae30041eba2ec8c2b4bdf460c370ef58d7a4df0b44e561520809b2d1e3099a68f99b2cc80885e4d4d33678802311b020400007b0203761f5e29944d79d76656323f106cf2efbf5f09e90000620100010000000000011233126964a9bc193366e5a469b21395351d522a829343acc954e434e8060ed82f75fc394fedeee13563dded034aa3694877cec59571d13116d741f0becd538b1b02010190d62a32d1cc65aa3e80b567c8c0d3ca0f411e61030400001601038951fe5b3a10cc282b391d5027a371e28fcf442c",
"accounts":[{"currency":"eur","chain":"polygon","isVisible":true}]
}
```

I also successfully linked a 4337 wallet through Blocto app.

I'm not sure how to resolve this until we can test your product.

**Comment by kevtechi** [comment]
Great, you seem to be more confident that the fixes you made in April work then 👍. 

We don't have plans to integrate Monerium anymore.

Maybe we'll take a look at it again in the future.
